src/compose/app.ts

```diff
@@ -764,13 +779,30 @@ export class App {
 	private dependenciesMetForServiceKill(
 		targetApp: App,
 		availableImages: Image[],
+		servicePairs: Array<ChangingPair<Service>>,
+		strategy: string,
 	) {
-		// Don't kill any services before all images have been downloaded
-		return this.targetImagesReady(targetApp, availableImages);
+		// if kill|delete-then-download, don't kill any services
+		// before new images in target have been downloaded.
+		// Services are safe to kill if they are in both current and target
+		// even if the target image isn't yet ready.
```

Comment:
I don't understand why this is needed. Result of `dependenciesMetForServiceKill` is not used by `getStepsForStrategy` if the strategy is `kill-then-download` or `delete-then-download` so this code would be redundant. Also this doesn't seem like the most efficient alternative considering this function is called for every service.

Reply:
Oh, wait, sorry, you changed the `getStepsForStrategy` but function, but I still don't understand why